### ISSUE TITLE
Don't require empty directory for `sails new`

### ISSUE THREAD

**Issue by anthonator** [balderdashy/sails-generate-new#7]
I like to groom my project directory before generating a new project. Most of this is installing dependencies into ```node_modules```. I work on a lot of projects and installing my dependencies into the global namespace isn't practical. So even installing sails into my projects local ```node_modules``` prevents me from generating a new sails project. Currently I create my folder, groom the directory, generate a sails project in a directory inside my projects folder and then copy everything back. A little cumbersome.

**Comment by jedd-ahyoung** [comment]
+1 for this.

**Comment by sgress454** [comment]
I'm not quite following...why can't you do `sails new project` and then install your modules?  Can you elaborate a bit on your process?

**Comment by anthonator** [comment]
Because `sails` is in `node_modules` and not the global namespace. The fact that `node_modules` exists in the folder I'm trying to create a project in causes `sails new` to fail.

**Comment by sgress454** [comment]
How are you running `sails new`?  Do you have Sails installed globally?  In that case, if you all you want is to not have your new project linked to the global sails, you can just do:

```
sails new myProj && cd myProj && rm -rf node_modules && npm install
```

That might be a good *option* for `sails new`, where instead of using symlinks it actually runs `npm install`.  This is how it worked way back when, but it sure was slow, and upgrading apps in development was a pain.  

My resistance to allowing `sails new` in non-empty directories is twofold:

1. The current behavior prevents you from accidentally dumping a bunch of Sails scaffold files in your "Documents" directory (or wherever) that you then have to hunt and delete.
2. Even when done as intended, the files in the existing directory might prevent your new Sails app from running properly--for instance, you might already have a pre-installed *older* version of Sails in `node_modules` that conflicts with the files that were added by `sails new`.

I just think it's better to start clean and copy things in to the directory (like you've been doing), where you have full control over what gets moved and overwritten.

**Comment by cyrilchapon** [comment]
``` 
git clone http://gitserver/myrepo
cd myrepo
sails new .
```

That fails, and that's anoying =/

**Comment by sgress454** [comment]
@cyrilchapon What's in that repo?  Assets?

> Adding a warning and a --force argument would satistfy everyone

Agreed--sounds like a brilliant pull request for someone!

**Comment by cyrilchapon** [comment]
> @cyrilchapon What's in that repo? Assets?

Nothing at all, cloning is just a shortcut to init/add origin/set upstream here

**Comment by jetli13** [comment]
+1

**Comment by sailsbot** [comment]
Thanks for posting, @anthonator.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
